test/singletons.test.js

```diff
@@ -240,6 +240,37 @@ describe('singletons', function() {
     expect(allPrivs.length).to.equal(1);
   });
 
+  it('keeps server singleton even when a local item is created before sync', async function () {
+    const identifier  = this.application.identifier;
+    await this.registerUser();
+
+    /** Create prefs and associate them with account */
+    const ogPrefs = await insertPrefsPayload(this.application);
+    await this.application.sync(syncOptions);
+
+    /** Remove prefs locally and create a newer one */
+    this.application.lockSyncing();
+    await this.application.itemManager.removeItemLocally(ogPrefs);
+    const localPrefs = await insertPrefsPayload(this.application);
```

Comment:
If syncing is locked this value would never be persisted to the DB. So when you recreate the application `expect(this.application.findItem(localPrefs.uuid)).to.not.exist;` would always pass.

Reply:
Ah shoot, so my test may be bonked then. Is there an easy way to persist an item to storage without syncing that you know of?

Reply:
storageService.savePayload probably

Reply:
Just to be sure you should also add an assertion after you deinit that the item can be found in storage.

Reply:
Awesome, thanks!